File: 백준/Silver/2164. 카드2/카드2.cc
// Online C++ compiler to run C++ program online
#include <iostream>
#include <stack>
#include <queue>
using namespace std;
int N;

int main() {
    
    queue<int> q;
    cin >> N;
    
    for(int i=1;i<=N;i++)
    {
        q.push(i);
    }
    while(q.size()!=1)
    {
        q.pop();
        int f = q.front();
        q.pop();
        q.push(f);
    }
    cout << q.front();

    return 0;
}

File: 백준/Silver/2164. 카드2/카드2.swift
let n = Int(readLine()!)!
var queue: [Int] = []
var front = 0

for i in 1...n {
    queue.append(i)
}

while queue.count - front > 1 {
    front += 1
    queue.append(queue[front])
    front += 1
}

print(queue[front])